lxd/instance/drivers/driver_qemu.go

```diff
@@ -2866,17 +2866,43 @@ if [ ! -d "/run/systemd/system/" ]; then
     exit 1
 fi
 
+for path in "/lib/systemd" "/usr/lib/systemd"; do
+    [ -d "${path}/system" ] || continue
+    LIB_SYSTEMD="${path}"
+    break
+done
+
+if [ ! -d "${LIB_SYSTEMD:-}" ]; then
+    echo "Could not find path to systemd"
+    exit 1
+fi
+
+for path in "/lib/udev" "/usr/lib/udev"; do
+    [ -d "${path}/rules.d/" ] || continue
+    LIB_UDEV="${path}"
+    break
+done
+
+if [ ! -d "${LIB_UDEV:-}" ]; then
+    echo "Could not find path to udev"
+    exit 1
+fi
+
 # Cleanup former units.
-rm -f /lib/systemd/system/lxd-agent-9p.service \
-    /lib/systemd/system/lxd-agent-virtiofs.service \
+rm -f "${LIB_SYSTEMD}/system/lxd-agent-9p.service" \
+    "${LIB_SYSTEMD}/system/lxd-agent-virtiofs.service" \
     /etc/systemd/system/multi-user.target.wants/lxd-agent-9p.service \
     /etc/systemd/system/multi-user.target.wants/lxd-agent-virtiofs.service \
     /etc/systemd/system/multi-user.target.wants/lxd-agent.service
 
 # Install the units.
-cp udev/99-lxd-agent.rules /lib/udev/rules.d/
-cp systemd/lxd-agent.service /lib/systemd/system/
-cp systemd/lxd-agent-setup /lib/systemd/
+cp udev/99-lxd-agent.rules "${LIB_UDEV}/rules.d/"
+cp systemd/lxd-agent-setup "${LIB_SYSTEMD}/"
+if [ "/lib/systemd" = "${LIB_SYSTEM}" ]; then
+  cp systemd/lxd-agent.service "${LIB_SYSTEMD}/system/"
+else
+  sed "/=\/lib\/systemd/ s|=/lib/systemd|=${LIB_SYSTEMD}|" systemd/lxd-agent.service > "${LIB_SYSTEMD}/system/lxd-agent.service"
```

Comment:
what is this part for?

Reply:
I put a comment that we need to adapt the path when systemd is installed in the unusual location.

Reply:
Thanks, what is it actually doing though?

Reply:
Replacing `=/lib/systemd` by `=${LIB_SYSTEMD}` only if both are different.

Reply:
If `LIB_SYSTEMD` == `/lib/systemd`, `cp` is used. Otherwise, `sed` will take the `lxd-agent.service` file from the config and return a mangled version to stdout which is then written into the proper location (also based on `${LIB_SYSTEMD}`).

This way, `sed` is only used/needed if the unit needs mangling and there is no "in-place" rewrite as a `cp + sed -i` would have required.

Reply:
Thanks!